### PR TITLE
Register callback on stream end

### PR DESCRIPTION
**Pull request by taojang**
- Clean up unused code
- Try use retry library (unfortunately not all interesting functions are exported)
- Register callback when stream ends

### CODE REVIEW

File: src/Nakadi/Client/Stream.purs [hunk 1]
```diff
@@ -136,6 +133,7 @@ handleRequest
 handleRequest { resultVar, buffer, bufsize } streamParams resStream commit eventHandler env = do
   callback <- splitAtNewline buffer bufsize  handle
   onData resStream callback
+  onEnd resStream (launchAff_ $ AVar.put StreamClosed resultVar)
```
Comment (i-am-the-slime) [inline src/Nakadi/Client/Stream.purs]:
```suggestion
  onEnd resStream (launchAff_ $ AVar.put StreamClosed resultVar)
  onError resStream (\e -> do
    env.logWarn Nothing $ "Error in read stream " <> message e
    launchAff_ $ AVar.put StreamClosed resultVar
    )
```

Comment (taojang) [reply src/Nakadi/Client/Stream.purs]: It's logged in retry part of `Client.purs`

Comment (taojang) [reply src/Nakadi/Client/Stream.purs]: oops, sorry, you mean the error

Comment (i-am-the-slime) [reply src/Nakadi/Client/Stream.purs]: ah
so not?

**Review by i-am-the-slime — COMMENTED** [review]
🥇

**Pull request merged**